### PR TITLE
Egui Drag an Drop shards

### PR DESCRIPTION
**Pull request by guusw**
Opening a PR so I don't forget about this.

### CODE REVIEW

File: shards/core/into_wire.hpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#ifndef C81B8E95_5381_4841_8931_F7EE0CB8EEEA
+#define C81B8E95_5381_4841_8931_F7EE0CB8EEEA
+
+#include <shards/common_types.hpp>
+#include <shards/shards.hpp>
+
+namespace shards {
+
+// Converts either a Wire/ShardsSeq into a std::shared_ptr<SHWire>
+struct IntoWire {
```
Comment (guusw) [inline shards/core/into_wire.hpp:10]: TODO: Change this into a function

File: shards/modules/core/wires.cpp [hunk 2]
```diff
@@ -2061,12 +2061,14 @@ struct DoMany : public TryMany {
...
 
   static inline Parameters _params{
-      {"Wire", SHCCSTR("The wire to run many times sequentially."), WireBase::WireTypes},
+      {"Wire", SHCCSTR("The wire to run many times sequentially."), IntoWire::RunnableTypes},
   };
 
   static SHParametersInfo parameters() { return _params; }
 
   SHTypeInfo compose(const SHInstanceData &data) {
+    wire = IntoWire{}.runnable(wireref);
```
Comment (guusw) [inline shards/modules/core/wires.cpp:2070]: Also add to TryMany

File: shards/egui/src/util.rs [hunk 3]
```diff
@@ -63,7 +64,7 @@ where
   result
 }
 
-pub fn get_object_from_var_opt<'a, T>(
+pub unsafe fn get_object_from_var_opt<'a, T>(
```
Comment (sinkingsugar) [inline shards/egui/src/util.rs:67]: why unsafe?

File: shards/egui/src/util.rs [hunk 4]
```diff
@@ -74,14 +75,17 @@ pub fn get_object_from_var_opt<'a, T>(
   }
 }
 
-pub fn get_object_from_var<'a, T>(var: &Var, obj_type: &Type) -> Result<&'a mut T, &'static str> {
+pub unsafe fn get_object_from_var<'a, T>(
```
Comment (sinkingsugar) [inline shards/egui/src/util.rs:78]: why unsafe?

File: shards/egui/src/util.rs [hunk 5]
```diff
@@ -74,14 +75,17 @@ pub fn get_object_from_var_opt<'a, T>(
...
+  obj_type: &Type,
+) -> Result<&'a mut T, &'static str> {
   match get_object_from_var_opt(var, obj_type)? {
     Some(obj) => Ok(obj),
     None => Err("Object not set"),
   }
 }
 
-pub fn get_object_from_param_var<'a, T>(
+pub unsafe fn get_object_from_param_var<'a, T>(
```
Comment (sinkingsugar) [inline shards/egui/src/util.rs:88]: why unsafe?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]

File: shards/egui/src/util.rs (hunk 3 above)
Comment (guusw) [reply shards/egui/src/util.rs:67]: Because the type you're casting into could not be the type that is there.

Comment (guusw) [reply shards/egui/src/util.rs:67]: same for comments below

**Comment by guusw** [comment]
Added inline shard support for TryMany, StepMany & Expand + tests.

**Pull request merged**